pkg/util/util.go

```diff
@@ -41,9 +41,17 @@ func ToCanonicalName(name string) string {
 	return strings.Join(nameParts, ".")
 }
 
+// CanonicalTypeNamer is an interface for models without Go types to seed model name
+type CanonicalTypeNamer interface {
+	CanonicalTypeName() string
```

Comment:
upon consideration, this is a fairly vague name for a fairly specific piece of information. the other interface we let things implement is `OpenAPIDefinitionGetter` via `OpenAPIDefinition() *OpenAPIDefinition`. Starting the method with `OpenAPI` is probably a good idea for coherence:

`OpenAPICanonicalTypeName() string`

Reply:
Needs documentation about the expected (required?) format of the returned name. Examples are probably a good idea.

Reply:
updated. I don't have a good doc/spec that defines "REST friendly name" to refer to, so I just write down some example

I remember this util came from https://github.com/kubernetes/kubernetes/pull/40124/commits/239169a5a45d94b7f404c080743953fcc26f50b6#diff-1f607a313d799650deac9489e362de2cR131. @mbohlool do you have suggestion on how we document this?

Reply:
nvm, the namer has nothing to do with canonical (REST friendly) name above